dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/taglib/GlobalTagLibraryCache.java

```diff
@@ -127,7 +127,19 @@ public GlobalTagLibraryCache(ClassLoader loader) {
             tldParser = new TldParser(this, configManager, false);
             
             TagLibCacheConfigParser tagLibCacheConfigParser = new TagLibCacheConfigParser();
-            tagLibCacheConfigParser.parse(this.getClass().getResourceAsStream("/com/ibm/ws/jsp/resources/taglibcacheconfig.xml"));
+
+            if(com.ibm.ws.jsp.webcontainerext.JSPExtensionFactory.isPages30orHigher()){
+                if(com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable(Level.FINE)){
+                    logger.logp(Level.FINE, CLASS_NAME, "GlobalTagLibraryCache", "Parsing taglibcacheconfig.jstl.2.0.xml for pages version >= 3.0");
+                }
+                tagLibCacheConfigParser.parse(this.getClass().getResourceAsStream("/com/ibm/ws/jsp/resources/taglibcacheconfig.jstl.2.0.xml"));
+            } else {
+                if(com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable(Level.FINE)){
+                    logger.logp(Level.FINE, CLASS_NAME, "GlobalTagLibraryCache", "Parsing taglibcacheconfig.xml for pages version <= 2.3");
+                }
+                tagLibCacheConfigParser.parse(this.getClass().getResourceAsStream("/com/ibm/ws/jsp/resources/taglibcacheconfig.xml"));
+            }
+
```

Comment:
Parses the correct taglibcacheconfig file based on which pages version is loaded.